airflow/auth/managers/simple/views/auth.py

```diff
@@ -0,0 +1,84 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+
+from flask import redirect, request, session, url_for
+from flask_appbuilder import expose
+
+from airflow.auth.managers.simple.user import SimpleAuthManagerUser
+from airflow.configuration import conf
+from airflow.utils.state import State
+from airflow.www.app import csrf
+from airflow.www.views import AirflowBaseView
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleAuthManagerAuthenticationViews(AirflowBaseView):
```

Comment:
Building on a flask blueprint would be more future proof.

Reply:
Could you provide more guidance how to do it please? I am using `self.appbuilder.add_view_no_menu` to register the view to the environment and this function needs a `BaseView` as parameter

Reply:
Do you mean to do it like a [plugin](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html)? If so, auth managers are not built that way so it would require quite some modifications (including in the interface) to do that. I am wondering if it is worth it and it would no be better to wait for AIP-68 to be complete so that we can leverage it

Reply:
This is probably the most concise example:

https://github.com/apache/airflow/blob/4323b1dd7d3a6603b7cc15056f06f5bf2de0569d/airflow/www/extensions/init_react_ui.py#L24-L39

Reply:
The flask docs cover it pretty well too. You just need access to the flask app (which I think you can get back out of appbuilder - though that'll eventually change once we break FAB out).

Reply:
Happy to do it but I dont know if it is worth it. Blueprints are objects from Flask which will go away when we remove FAB. So basically I dont quite see the value of using a blueprint here since anyway we will have to revisit the way to register views from auth managers when AIP-68 is complete

Reply:
FAB removal doesn't necessarily mean flask removal too. Prior to this PR where it came up, I don't think it was even mentioned... not saying we can't though.

Reply:
Feel free to leave it as-is though. Easy enough to refactor when I get around to doing the rest of the FAB stuff.